### ISSUE TITLE
Type-checking issue for ternary (`?:`) operator

### ISSUE THREAD

**Issue by tangent-vector** [csyonghe/Spire#29]
This code:

```
float a, b, c;
a = true ? b : c;
```

Yields the error message:

```
error 30080: the two value expressions in a select clause must evaluate to same type.
```

**Comment by csyonghe** [comment]
This is fixed in latest code but I can't push now, to fix in your branch, search "Error(30080, L"the two value"
and modify the if statement above it to:

if (!expr->Expr0->Type->Equals(expr->Expr1->Type))

**Comment by tangent-vector** [comment]
I worked around this for now by using an `if` statement. If I find that I need the fix I will apply it. Thanks!

**Comment by tangent-vector** [comment]
Tried applying that fix, and it seems to fail on implicit conversion of the `expr->Expr1->Type` part to an ordinary pointer (it is a smart pointer). I inserted an explicit call to `.Ptr()` and it was fine, so I'm not sure what is the reason for the issue.

Also note that I appear to have run into a suspiciously similar issue with Spire's range-style `for` loop where I get an `error 30019: type mismatch 'int' and 'int'`. Looking at the code of `VisitForStatement` there appear to be similar type `!=` tests that need to be changed to use `Equals`.
